engine/runtime-integration-tests/src/test/scala/org/enso/interpreter/test/instrument/RuntimeAsyncCommandsTest.scala

```diff
@@ -580,6 +612,20 @@ class RuntimeAsyncCommandsTest
     context.send(
       Api.Request(requestId, Api.PushContextRequest(contextId, item1))
     )
+    var isProgramStarted  = false
+    var iteration         = 0
+    var out: List[String] = Nil
+    while (!isProgramStarted && iteration < 100) {
```

Comment:
the whole `while` cycle could be in `synchronized` block - preferrably in a dedicated `awaitXyz` method.